ODIN_II/SRC/netlist_create_from_ast.cpp

```diff
@@ -3469,51 +3473,129 @@ signal_list_t* create_operation_node(ast_node_t* op, signal_list_t** input_lists
 
-            /* hookup the input pins = will do padding of zeros for smaller port */
-            hookup_input_pins_from_signal_list(operation_node, current_idx, input_lists[i], 0, input_port_width, verilog_netlist);
-            current_idx += input_port_width;
         } else {
-            /* ELSE if taking the port widths as they are */
-            /* allocate the pins needed */
-            allocate_more_input_pins(operation_node, input_lists[i]->count);
-            /* record this port size */
-            add_input_port_information(operation_node, input_lists[i]->count);
+            /* record the size of the shift */
+            int shift_size = op->children[1]->types.vnumber->get_value();
+
+            switch (operation_node->type) {
+                case SL:
+                case ASL: {
+                    /* connect ZERO to outputs that don't have inputs connected */
+                    for (i = 0; i < shift_size; i++) {
+                        // connect 0 to lower outputs
+                        npin_t* zero_pin = allocate_npin();
+                        add_fanout_pin_to_net(verilog_netlist->zero_net, zero_pin);
+
+                        add_pin_to_signal_list(return_list, zero_pin);
+                        zero_pin->node = NULL;
+                    }
 
-            /* hookup the input pins */
-            hookup_input_pins_from_signal_list(operation_node, current_idx, input_lists[i], 0, input_lists[i]->count, verilog_netlist);
+                    /* connect inputs to outputs */
+                    for (i = 0; i < output_port_width - shift_size; i++) {
+                        // connect higher output pin to lower input pin
+                        add_pin_to_signal_list(return_list, input_lists[0]->pins[i]);
+                        input_lists[0]->pins[i]->node = NULL;
+                    }
+                    break;
+                }
+                case SR: {
+                    for (i = shift_size; i < output_port_width; i++) {
+                        // connect higher output pin to lower input pin
+                        add_pin_to_signal_list(return_list, input_lists[0]->pins[i]);
+                        input_lists[0]->pins[i]->node = NULL;
+                    }
+
+                    /* connect ZERO to outputs that don't have inputs connected */
+                    for (i = output_port_width - 1; i >= output_port_width - shift_size; i--) {
+                        // connect 0 to lower outputs
+                        npin_t* zero_pin = allocate_npin();
+                        add_fanout_pin_to_net(verilog_netlist->zero_net, zero_pin);
+
+                        add_pin_to_signal_list(return_list, zero_pin);
+                        zero_pin->node = NULL;
+                    }
+                    break;
+                }
+                case ASR: {
+                    error_message(NETLIST, op->loc, "%s", "TODO: test that ASR works with signed number\n");
```

Comment:
please add tests for that.

Reply:
Already some tests called asr_x_x exist in benchmarks!

Reply:
then update that comment to reflect the status.